cpu/stm32f4/startup.c

```diff
@@ -41,6 +41,11 @@ extern void kernel_init(void);
 extern void __libc_init_array(void);
 
 /**
+ * Required by g++ cross compiler
+ */
+void* __dso_handle;
```

Comment:
This is probably required for other platforms as well. I have problems explaining why and when exactly it is needed, but my compiler complains otherwise. I will try to find out more ...